libsupport/include/katana/EntityTypeManager.h

```diff
@@ -17,10 +17,16 @@
 
 namespace katana {
 
-/// EntityTypeID uniquely identifies an entity (node or edge) type
-/// EntityTypeID for nodes is distinct from EntityTypeID for edges
-/// This type may either be an atomic type or an intersection of atomic types
-/// EntityTypeID is represented using 8 bits
+/// EntityTypeID uniquely identifies an entity (node or edge) type, which is
+/// a set of strings (labels).
```

Comment:
Label and type can conflate with each other. Cypher uses "node labels" and "edge types". We call both of them as "atomic types". Instead of this, I'd prefer something like:

```cpp
/// EntityTypeID uniquely identifies an entity (node or edge) type, which is
/// a set of user-defined strings.
/// Types may either be an atomic type or an intersection of atomic types,
/// where an atomic type is a unique user-defined string and 
/// intersection of atomic types is a set of corresponding user-defined strings.
/// The EntityTypeID space for nodes is distinct from EntityTypeID for edges.
///
/// Types are sets of strings; all other representations are optimizations.  Clients
/// should never compare EntityTypeIDs across different managers.  Only comparison of
/// the string representations are valid.
///
/// EntityTypeID is represented using 16 bits
```